solr/solrj/src/java/org/apache/solr/client/solrj/io/stream/AlertStream.java

```diff
@@ -0,0 +1,156 @@
+
+  TupleStream tupleStream;
+  String targetURL;
+
+  public AlertStream(StreamExpression expression, StreamFactory factory) throws IOException {
+    List<StreamExpression> streamExpressions = factory.getExpressionOperandsRepresentingTypes(expression, Expressible.class, TupleStream.class);
+
+    String targetURL = null;
+
+    // parameters by position
+    TupleStream tupleStream = factory.constructStream(streamExpressions.get(0));
+    StreamExpressionNamedParameter urlExpression = factory.getNamedOperand(expression, "url");
+
+    if (urlExpression != null) {
+      targetURL = ((StreamExpressionValue) urlExpression.getParameter()).getValue();
+    }
+
+    init(tupleStream, targetURL);
+  }
+
+  public void init(TupleStream tupleStream, String targetURL) {
+    this.tupleStream = tupleStream;
+    this.targetURL = targetURL;
+  }
+
+  public AlertStream(TupleStream tupleStream, String targetURL) {
+    init(tupleStream, targetURL);
+  }
+
+  @Override
+  public void setStreamContext(StreamContext context) {
+    tupleStream.setStreamContext(context);
+  }
+
+  @Override
+  public List<TupleStream> children() {
+    ArrayList<TupleStream> list = new ArrayList<>();
+    list.add(tupleStream);
+    return list;
+  }
+
+  @Override
+  public void open() throws IOException {
+    tupleStream.open();
+  }
+
+  @Override
+  public void close() throws IOException {
+    tupleStream.close();
+  }
+
+  @Override
+  public Tuple read() throws IOException {
+    Tuple tuple = tupleStream.read();
+
+    if (!tuple.EOF) {
+      alert(tuple);
+    }
+    return tuple;
+  }
+
+  @Override
+  public StreamComparator getStreamSort() {
+    return tupleStream.getStreamSort();
+  }
+
+  @Override
+  public StreamExpressionParameter toExpression(StreamFactory factory) throws IOException {
+    StreamExpression expression = new StreamExpression(factory.getFunctionName(this.getClass()));
+    expression.addParameter(((Expressible) tupleStream).toExpression(factory));
+    return expression;
+  }
+
+  @Override
+  public Explanation toExplanation(StreamFactory factory) throws IOException {
+    return new StreamExplanation(getStreamNodeId().toString())
+      .withChildren(new Explanation[]{
+        tupleStream.toExplanation(factory)
+      })
+      .withFunctionName(factory.getFunctionName(this.getClass()))
+      .withImplementingClass(this.getClass().getName())
+      .withExpressionType(Explanation.ExpressionType.STREAM_DECORATOR)
+      .withExpression(toExpression(factory).toString());
+  }
+
+  void alert(Tuple tuple) throws IOException {
+    try {
+      Map<String, Map> requestBody = new HashMap<String, Map>();
+      requestBody.put("matches", tuple.fields);
+
+      CloseableHttpClient httpClient = HttpClients.createDefault();
+      HttpPost httpPost= new HttpPost((URI.create(targetURL)));
+      httpPost.setEntity(new StringEntity(requestBody.toString()));
+
+      httpClient.execute(httpPost);
```

Comment:
Sorry I'm not sure which tuple do you mean. Right now the alert request is sent to the endpoint for each matching tuple so users could retrieve the HTTP status for each request received at the endpoint. 

I added the timestamp for sending each request in the tuple.

Reply:
when adding the timestamp, you should also add the HTTP status code and maybe other information so that the users know if certain documents were not processed properly